src/domain/tagset/tagset.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable } from '@nestjs/common';
+import { ITagsetable } from 'src/interfaces/tagsetable.interface';
+import { RestrictedTagsetNames, Tagset } from './tagset.entity';
+import { ITagset } from './tagset.interface';
+
+@Injectable()
+export class TagsetService {
+
+  // Helper method to ensure all members are initialised properly.
+  // Note: has to be a seprate call due to restrictions from ORM.
+  async initialiseMembers(tagset: ITagset): Promise<ITagset> {
+    if (!tagset.tags) {
+      tagset.tags = [];
+    }
+
+    return tagset;
+  }
+
+  async createRestrictedTagsets(tagsetable: ITagsetable, names: string[]): Promise<boolean> {
+    if (!tagsetable.restrictedTagsetNames) {
+      throw new Error('Non-initialised tagsetable submitted');
+    }
+    for (const name of names) {
+      const tagset = new Tagset(name);
```

Comment:
createRestrictedTagsets is a very odd method. You iterate over names, create  a new Tagset and never use it. I presume that's not the idea.